ocaml/xapi-storage-script/main.ml

```diff
@@ -1709,16 +1711,38 @@ let bind ~volume_script_dir =
     set ~dbg ~sr ~vdi ~key:_vdi_type_key ~value:"cbt_metadata"
   in
   S.VDI.data_destroy vdi_data_destroy_impl ;
+  let vdi_compose_impl dbg sr parent child =
+    wrap
+    @@
+    let* sr = Attached_SRs.find sr in
+    let child = Storage_interface.Vdi.string_of child in
+    let parent = Storage_interface.Vdi.string_of parent in
+    return_volume_rpc (fun () ->
+        Volume_client.compose (volume_rpc ~dbg) dbg sr child parent
+    )
+  in
+  S.VDI.compose vdi_compose_impl ;
+  let vdi_set_content_id_impl dbg sr vdi content_id =
+    wrap
+    @@
+    let* sr = Attached_SRs.find sr in
+    let vdi = Storage_interface.Vdi.string_of vdi in
+    let* () = set ~dbg ~sr ~vdi ~key:_vdi_content_id_key ~value:content_id in
+    return ()
+  in
+  S.VDI.set_content_id vdi_set_content_id_impl ;
+  let vdi_add_to_sm_config_impl _dbg _sr _vdi _key _value = wrap @@ return () in
+  S.VDI.add_to_sm_config vdi_add_to_sm_config_impl ;
+  let vdi_remove_from_sm_config_impl _dbg _sr _vdi _key = wrap @@ return () in
+  S.VDI.remove_from_sm_config vdi_remove_from_sm_config_impl ;
```

Comment:
Why are these empty?

Reply:
They shouldn't be. I havfe updated these